nemo_aligner/models/nlp/gpt/megatron_gpt_reward_model.py

```diff
@@ -350,93 +353,63 @@ def finish_validation_step(self):
         finish_validation_step(self)
 
     def infer(
-        self,
-        inputs: Union[List[str], List[List[int]]] = None,
-        sequence_length: List[int] = None,
-        add_EOS: bool = False,
+        self, inputs: Union[List[str], torch.Tensor, List[dict]], add_BOS=False, add_EOS=False,
```

Comment:
I kept it in case the user wants to customize the `tokenize_batch` function below. But I'm not too adamant on either way. Do you think we should delete this?(and i guess add_BOS as well)

Reply:
Right now it's all a bit confusing IMO:
* serve_reward_model [hardcodes both BOS & EOS to False](https://github.com/NVIDIA/NeMo-Aligner/blob/efadcaec90bb5b966286f54653e38aa7e5552dee/examples/nlp/gpt/serve_reward_model.py#L58)
* `RewardModelDataset` allows [adding EOS](https://github.com/NVIDIA/NeMo-Aligner/blob/2421c7354232bb08f89100ed2c297eae938d5ee9/nemo_aligner/data/nlp/datasets.py#L182) (but not BOS)
* This `infer()` method allows to add either BOS or EOS

=> I think we should be consistent in what is supported. I'm ok for this to happen in a future PR but let's prioritize it to be sure this gets addressed sooner than later.
(or, we can remove BOS & EOS support in all RM code in this PR, and consider adding it back later if there's a need for it)

Reply:
yeah this might be good for a future PR. I think what should happen is we should support add_BOS and add_EOS everywhere and default it(rather than hardcode) it to False

Reply:
I raised it in this issue #231 just so it doesn't get loss into the abyss

Reply:
Sounds good, thanks